### ISSUE TITLE
Receive less logs than registered in phones log

### ISSUE THREAD

**Issue by leonardoporpora** [guardianproject/haven#232]
I receive much less messages through signal than the logs captured on the phone. I receive 2 photos but on the phone there were 16 photos

**Comment by leonardoporpora** [comment]
I’m using the latest beta

**Comment by lukeswitz** [comment]
Try changing setting "Notification Time Interval" to 0. Restart and see if that does it.

**Comment by leonardoporpora** [comment]
It’s not this the issue I have tried but nothing changed

**Comment by n8fr8** [comment]
So, even if you set the internal to 0, you still don't receive all the notifications?

**Comment by leonardoporpora** [comment]
Exactly

**Comment by leonardoporpora** [comment]
@n8fr8 I have tried again and I got 20 logs in the phone and only 3 photos I have received

**Comment by n8fr8** [comment]
It could be that Signal might be throttling the amount of content coming through. I will look into it.

**Comment by leonardoporpora** [comment]
Thanks

**Comment by leonardoporpora** [comment]
@n8fr8 have you found anything interesting?

**Comment by leonardoporpora** [comment]
hey @n8fr8 have u found anything?

**Comment by leonardoporpora** [comment]
still have the same issue

**Comment by leonardoporpora** [comment]
i have received just 3 logs but on the phone there were 28..... @lukeswitz

**Comment by lukeswitz** [comment]
Let's see how it goes with an update to the signal-cli. Will report back.

**Comment by leonardoporpora** [comment]
thank :)

**Comment by leonardoporpora** [comment]
@lukeswitz take a look at this https://whispersystems.discoursehosting.net/t/signal-rate-limits-any-documentation/1223/3

**Comment by lukeswitz** [comment]
Good find, but there isn't any limiting going on using the guardianproject/signal-cli-android that I can see. It handles a pretty intense flow of messages when it's active:

A 'message bomb' will work, but not on the first run. You can try testing using:
- fresh launch, 
- recording a dummy session with at least one alert
- next session should hit each event. 

Thanks for your help. Will update if things work better on the new branch.

**Comment by lukeswitz** [comment]
I found none haven-side, but the Signal app does use limiting. Signal logs indicate rate limit exceeded upon the first attempt to register a number if Haven has been used previously/in parallel. 

It's possible the acceptance rate of messages in Signal follows this standard, which makes sense. It will lock users out from registering devices without user feedback (as I've found in multi-device support tests).

**Comment by leonardoporpora** [comment]
But instead of kill the upload of the photo it could make a queue, is it possible so that i would receive all photos?

**Comment by leonardoporpora** [comment]
I have still the same problem and this is a bad issue cause I saw that there were 65 events on my phone but I received only 7 photos in which none where shown, seeing the logs I had a person that I could recognize... But if he takes my phone and I haven't received the photos on my primary phone ...

**Comment by lukeswitz** [comment]
Same issues persisting here across all devices tested. The problem isn’t with Signal, Haven is asking a bit too much of the CLI integration. 

Signal docs mention that messages aren’t completely independent when sent in a small window, which could easily explain why 60 are coming through as only six. 

Perhaps event logging needs an asynchronous method to handle the db entry & alerts (as bitmap conversion footprint is currently higher than any prior release) 

Are you able to view video events remote/local? I noted adding even 10ms latency improved continuity across logs and alerts for images. 

Lots of opportunities across the board to rewrite the persistence layer and solve issues on legacy & flagship devices. Thanks again for continued feedback!

**Comment by n8fr8** [comment]
I've opened a new ticket to address new feature work to implement a complete remote access/sync mechanism

#370 

Signal is just meant for notification, not full remote backup
